packages/core/src/lib/collapse.svelte

```diff
@@ -0,0 +1,94 @@
+<!--
+@component
+  
+An element that Toggles visibility of content.
+
+```svelte
+    <Collapse
+      title='Motor 1'
+      variant="minimal"
+    >
+      <Breadcrumbs slot='title' crumbs={['Robot', 'Motor']}></Breadcrumbs>
+      <Badge slot='header' label='Inactive'></Badge>
+      <div style='font-size: 12px; padding: 1rem'>Motor one was concieved and executed at Bell Labs in 1972 under the guidance of lead director Dennis Richie and Superviser Wallace Breen.</div>
+    </Collapse>
+```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+import { createEventDispatcher } from 'svelte';
+import { Icon } from '$lib';
+
+type Variants = 'default' | 'minimal';
+
+/**
+ * The title for the collapse component.
+ */
+export let title = '';
+/**
+ * Whether the collapse is in the open position.
+ */
+export let open = false;
+/**
+ * The variant of the collapse component.
+ */
+export let variant: Variants = 'default';
+
+const dispatch = createEventDispatcher();
+
+const handleClick = () => {
+  open = !open;
+  dispatch('toggle', { isOpen: open });
+};
+</script>
+
+<div class="relative w-full">
+  <div
+    role="button"
+    aria-label="Toggle Content"
+    tabindex="0"
+    class="{cx(
+      'w-full py-2 px-4 flex flex-reverse items-center justify-between text-default cursor-pointer',
+      {
+        'border border-light bg-white': variant === 'default',
+      }
+    )},"
+    on:click={handleClick}
+    on:keyup|preventDefault={handleClick}
+  >
+    <div class="flex flex-wrap gap-x-3 gap-y-1 items-center">
+      {#if title}
+        <h2 class="m-0 text-sm">{title}</h2>
+      {/if}
+
+      <slot name="title" />
+    </div>
+
+    <div class="h-full flex items-center gap-3">
+      <slot name="header" />
+
+      <div
+      class={cx(
+        'transition-transform duration-200',
+        {
+          'rotate-0': !open,
+          'rotate-180': open,
+        }
+      )}
+    >
+      <Icon name="chevron-down" />
+    </div>
+    </div>
+  </div>
+
+  <div
+    class={cx('text-black transition-all duration-500', {
+      'bg-white': variant === 'default',
+      hidden: !open,
+    })}
+  >
+    <slot />
```

Comment:
I'd like to change how we handle this slot as we revise `<Collapse>` - specifically moving to conditionally rendering this slot rather than making it hidden. Before we had prime-core I created a revised version of collapse [here](https://github.com/viamrobotics/rdk/blob/main/web/frontend/src/lib/components/collapse.svelte).

Would you be up for moving the slot changes here? I'd like to work toward removing most of the RDK component and replacing it with this one.

The optimization is mostly for performance; rendering fewer nodes, especially if they're not used, almost always means better performance.

Reply:
There is also some local storage stuff but I have doubts that it belongs in Prime, we may just have to make a wrapper component for <Collapse> in RDK that persists state.

Reply:
Absolutely, will update accordingly